### PR TITLE
Chore/sc 37016 write registration tests

### PR DESCRIPTION
**Pull request by Tatiinzhee**
Added registration happy pass test.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,7 +3,9 @@
 
  npm 8
  - npm install
- - npx playwright install chromium webkit 
+ - npx playwright install chromium webkit
+ - npm i dotenv
+ - npm install @faker-js/faker --save-dev
```
Comment (mila-qa) [inline README.md:8]: npm install will install dotenv and faker. No need to specify every package

File: framework/pages/registration.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+const registerWithEmailButton = 'register-with-email-button';
+const firstnameField = 'firstName-input';
+const lastmameField = 'lastName-input';
+const emailaddressField = 'email-input';
+const passwordField = 'password-input';
+const continueButton = 'form-registration-submit';
+const clickRegisterButton = async (page: Page, isMobile: Boolean) => {
+    if(isMobile){
+        await clickOnElement(page, 'content-sidebar-toggle');
```
Comment (mila-qa) [inline framework/pages/registration.ts:18]: If we declare selectors as constants, please let's not mix both approaches. Let's move selectors to constants as you did for Email etc field

File: framework/pages/registration.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        await clickOnElement(page, 'activation-institution-input')
+        await fillInInput(page, 'activation-institution-input', 'Delft');
+        await clickOnElement(page, 'activation-institution-result-0');
+        await clickOnElement(page, 'university-year-3');
+        await clickOnElement(page, 'activation-submit-first-step-button');
+        await clickOnElement(page, 'activation-study-autocomplete-input');
+        await fillInInput(page, 'activation-study-autocomplete-input','econom');
+        await clickOnElement(page, 'activation-study-autocomplete-result-0');
+        await clickOnElement(page, 'activation-submit-second-step-button');
+        await waitForVisible(page, 'choose-premium-button');
```
Comment (mila-qa) [inline framework/pages/registration.ts:51]: it'd be nice to make a few function here. For example fillInregistrationFileds(), submitRegistrationModal ... .They will be reusable in many other tests

File: tests/registration.spec.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+const firstname = faker.name.firstName();
+const lastmame = faker.name.lastName();
+const emailaddress = faker.internet.email();
+const password = faker.internet.password();
+
+
+test.describe('Registration',  () => {
+  
+    test ('Successful registration', async ({page, isMobile}) => {
+        await app().Registration().registerWithEmailAndActivate(page, firstname, lastmame, emailaddress, password);
```
Comment (mila-qa) [inline tests/registration.spec.ts:16]: to follow up previous comment. If the registration page will have a few functions we can use them here to extend a bit test with steps on how exactly registration is going on. Similar to regular test case steps

**Review by mila-qa — CHANGES_REQUESTED** [review]

File: framework/pages/home.ts [hunk 5]
```diff
@@ -1,11 +1,11 @@
...
 
 const url = process.env.BASE_URL + process.env.BASE_LOCALIZATION;
 
 const openHome = async (page: Page) => {
     await page.goto(url);
-    await waitForVisible(page, 'the-cookie-banner');
+    await waitForVisible(page, 'the-cookie-banner'); 
 }
 
 const closeCookiesDisclaimer = async (page: Page) => {
```
[the remark this answers is not in the document]
Comment (danielcastro-studocu) [reply framework/pages/home.ts:11]: `clickloginButton` was moved to `login` page, so it should be removed from here.

File: package.json [hunk 6]
```diff
@@ -5,11 +5,14 @@
   "packages": {},
   "type": "module",
   "devDependencies": {
-    "@playwright/test": "^1.23.4",
-    "@types/node": "^18.7.11",
-    "dotenv": "^16.0.1"
+    "@faker-js/faker": "^7.5.0",
+    "@playwright/test": "^1.25.2",
+    "@types/mocha": "^9.1.1",
```
Comment (danielcastro-studocu) [inline package.json:10]: What are we using `mocha` for in this project?

File: package.json [hunk 7]
```diff
@@ -5,11 +5,14 @@
...
+    "@types/node": "^18.7.17",
+    "dotenv": "^16.0.2"
   },
   "scripts": {
+    "test": "playwright-tests-sandbox specs/**/*.spec.ts -timeout 50000 --require @babel/register",
```
Comment (danielcastro-studocu) [inline package.json:15]: What’s the purpose of this script? I got `sh: playwright-tests-sandbox: command not found` error.

File: tests/login.spec.ts [hunk 8]
```diff
@@ -1,7 +1,7 @@
 import {test} from '@playwright/test';
-import {clickOnElement, fillInInput, waitForVisible} from '../utils.ts';
-import * as home_page from '../pages/home.ts';
-import * as authentication_modal from '../pages/login.ts';
+import {clickOnElement, fillInInput, waitForVisible} from '../framework/helpers/utils.ts';
```
Comment (danielcastro-studocu) [inline tests/login.spec.ts:2]: `clickOnElement` and `fillInInput` are not used here, so you can remove them.

File: tests/registration.spec.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
+import app from '../framework/pages/index.ts';
+import { faker } from '@faker-js/faker';
+import { test, expect } from '@playwright/test';
+import { webkit } from '@playwright/test';
+
+
+const firstname = faker.name.firstName();
+const lastmame = faker.name.lastName();
```
Comment (danielcastro-studocu) [inline tests/registration.spec.ts:8]: Typo on `lastmame`.

File: tests/registration.spec.ts [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+const emailaddress = faker.internet.email();
+const password = faker.internet.password();
+
+const university = 'Delft';
+const study = 'econom';
+
+test.describe('Registration',  () => {
```
Comment (danielcastro-studocu) [inline tests/registration.spec.ts:15]: It’s a nice approach for grouping tests, but I think that different `specs` files will do that in the same way…

File: README.md [hunk 11]
```diff
@@ -17,6 +18,7 @@
 
  - click method doesn't check element state(visibility and so on) in spite of an announcement in documentation https://playwright.dev/docs/actionability
 
+ - for creating page fields/selectors we age going to use 'data-test-selector' plese refer to util.ts file to find out
```
Comment (danielcastro-studocu) [inline README.md:21]: Typo on `we age going to use` and `plese refer to`.

File: framework/pages/index.ts [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import RegisrtationPage from './registration.ts';
```
Comment (danielcastro-studocu) [inline framework/pages/index.ts:1]: Typo on `RegisrtationPage`.

**Review by danielcastro-studocu — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Tatiinzhee) [reply README.md]: Done

File: framework/pages/registration.ts (hunk 2 above)
Comment (Tatiinzhee) [reply framework/pages/registration.ts]: Done

File: framework/pages/registration.ts (hunk 3 above)
Comment (Tatiinzhee) [reply framework/pages/registration.ts]: Done

File: framework/pages/home.ts [hunk 13]
```diff
@@ -1,11 +1,11 @@
-import {clickOnElement, waitForVisible} from "../utils.ts";
+import {clickOnElement, waitForVisible} from "../helpers/utils.ts";
 import {type Page, expect} from "@playwright/test";
 
 const url = process.env.BASE_URL + process.env.BASE_LOCALIZATION;
 
 const openHome = async (page: Page) => {
     await page.goto(url);
-    await waitForVisible(page, 'the-cookie-banner');
+    await waitForVisible(page, 'the-cookie-banner');
```
[the remark this answers is not in the document]
Comment (Tatiinzhee) [reply framework/pages/home.ts]: done

File: framework/pages/login.ts [hunk 14]
```diff
@@ -0,0 +1,32 @@
+import {type Page} from "@playwright/test";
+import {clickOnElement, fillInInput, waitForVisible} from "..framework/helpers/utils.ts";
```
[the remark this answers is not in the document]
Comment (Tatiinzhee) [reply framework/pages/login.ts:2]: Sure, thx.

File: framework/pages/login.ts [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+    await clickOnElement(page, 'email-input', {delay: 1500, force: true});
+    await fillInInput(page, 'email-input', userName);
+    await fillInInput(page, 'password-input', userPwd);
+    await clickOnElement(page, 'submit-login-button');
+    await waitForVisible(page, 'user-menu-button');
+}
+
+export {
+    loginWithEmail
+}
```
[the remark this answers is not in the document]
Comment (Tatiinzhee) [reply framework/pages/login.ts:32]: Login test will be refactored within my next pr.

File: tests/login.spec.ts (hunk 8 above)
Comment (Tatiinzhee) [reply tests/login.spec.ts:2]: the login test will refactored in my next pr, thx

File: tests/registration.spec.ts (hunk 10 above)
Comment (Tatiinzhee) [reply tests/registration.spec.ts]: sure, thx

File: framework/pages/activation.ts [hunk 16]
```diff
@@ -0,0 +1,48 @@
+import {clickOnElement, waitForVisible, fillInInput} from "../helpers/utils.ts";
+import {type Page, expect} from "@playwright/test";
+
+const baseURL = process.env.BASE_URL;
+const local = process.env.BASE_LOCALIZATION;
+
+const url = baseURL + local;
```
[the remark this answers is not in the document]
Comment (Tatiinzhee) [reply framework/pages/activation.ts:7]: Let' keep by now, please, I still need to refactor the creating data.

**Review by danielcastro-studocu — CHANGES_REQUESTED** [review]
The only missing fix is the Typo on `RegisrtationPage`.

**Review by danielcastro-studocu — APPROVED** [review]

File: tests/registration.spec.ts (hunk 4 above)
Comment (mila-qa) [reply tests/registration.spec.ts]: as we agreed with @Tatiinzhee she will create steps for successful registration scenario in next PR

File: framework/pages/registration.ts (hunk 3 above)
Comment (mila-qa) [reply framework/pages/registration.ts]: dealing with home/landing page, accepting cookies etc will be moved into steps of test in related .spec.ts file. So we will have a ready test scenario in the spec file.
 Also registration and activation can be split even into smaller functions and they can be used during creating other test scenarios.

**Review by mila-qa — APPROVED** [review]
for now all good, suggested improvements are expected in upcoming PRs

**Pull request merged**